ci-operator/step-registry/baremetal/lab/agent/install/baremetal-lab-agent-install-commands.sh

```diff
@@ -293,14 +319,16 @@ for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
     # Assuming HTTP or HTTPS
     iso_path="${transfer_protocol_type}://${AUX_HOST}/${CLUSTER_NAME}.${arch}.iso"
   fi
-  # TODO: make the next one multi-threaded
   mount_virtual_media "${bmc_address}" "${redfish_user}" "${redfish_password}" \
-   "${iso_path}" "${transfer_protocol_type}" || {
-    echo "Failed to mount the ISO image in the host ${name}"
-    exit 1
-  }
+   "${iso_path}" "${transfer_protocol_type}" &
 done
 
+wait
+if [ -f /tmp/virtual_media_mount_failed ]; then
+  echo "Failed to mount the ISO image in one or more hosts"
+  exit 1
+fi
+
 for bmhost in $(yq e -o=j -I=0 '.[]' "${SHARED_DIR}/hosts.yaml"); do
```

Comment:
`# shellcheck disable=SC2154` before the for loop to fix the failing shell check test